dungeon/test/contrib/entities/ChestTest.java

```diff
@@ -17,42 +17,39 @@
 import java.io.IOException;
 import java.util.Optional;
 import java.util.Set;
-import org.junit.After;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
 
 /** Tests for the Chest classes. */
 public class ChestTest {
 
   /** WTF? . */
-  @After
+  @AfterEach
   public void cleanup() {
     Game.removeAllEntities();
     Game.currentLevel(null);
     Game.removeAllSystems();
   }
 
   /** Checks the correct creation of the Chest. */
+  @Disabled
   @Test
   public void checkCreation() throws IOException {
     Set<Item> itemData = Set.of();
     Point position = new Point(0, 0);
     Entity c = null;
     c = EntityFactory.newChest(itemData, position);
 
-    assertTrue(
-        "Needs the AnimationComponent to be visible to the player.",
-        c.fetch(DrawComponent.class).isPresent());
+    assertTrue(c.fetch(DrawComponent.class).isPresent());
     Optional<InventoryComponent> inventoryComponent = c.fetch(InventoryComponent.class);
-    assertTrue("Needs the InventoryComponent to be a chest", inventoryComponent.isPresent());
+    assertTrue(inventoryComponent.isPresent());
     assertEquals(
-        "Chest should have the given Items",
         new Item[] {null, null, null, null, null, null, null, null, null, null, null, null},
         inventoryComponent.get().items());
```

Comment:
@AMatutat Dieser Test schlägt nach dem Wechsel auf JUnit5 auf einmal fehl.

Reply:
Kann es irgendwie sein, dass das `assertEquals` in JUnit4 mit `equals` arbeitet und in JUnit5 mit `==`? Anders kann ich mir das irgendwie grad nicht erklären.

Reply:
hmmm. ich verstehe immer noch nicht, wieso man arrays mit dem normalen `assertEquals` vergleicht. sobald ich die korrekte methode nehme (`assertArrayEquals`), dann geht es auch :)

Reply:
hm...

- die `message` sollte das dritte Argument sein [(a)](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertEquals(java.lang.Object,java.lang.Object,java.lang.String)),
- und junit 5 hat eine `assertArrayEquals` Methode [(b)](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertArrayEquals(java.lang.Object%5B%5D,java.lang.Object%5B%5D,java.lang.String)).

Reply:
fixed in 31391ea3c284b2136c0aedbb775a719caf67c227

Reply:
> hm ...
> 
> * die `message` sollte das dritte Argument sein [(a)](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertEquals(java.lang.Object,java.lang.Object,java.lang.String)),
> * und JUnit 5 hat eine `assertArrayEquals`-Methode [(b)](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertArrayEquals(java.lang.Object%5B%5D,java.lang.Object%5B%5D,java.lang.String)).

@tgrothe vielen dank für den hinweis. 

(a) war hier nicht der punkt. das hatte ich bereits "umgebaut" bzw. behoben. 

und (b) galt ja auch bereits für junit4 ... ich frage mich nur, warum (1) das bisher trotzdem funktioniert hat und (2) warum ihr damals nicht einfach die korrekte methode genutzt habt.

Reply:
> ich frage mich nur, warum (1) das bisher trotzdem funktioniert hat und (2) warum ihr damals nicht einfach die korrekte methode genutzt habt.

Ich vermute, das Gleiche wie zum Beispiel bei Strings ... == wird bei gleichen und identischen String-Objekten true liefern, muss es bei nur gleichen Objekten aber nicht. equals muss bei gleichen Objekten aber immer true liefern, und vermutlich gibt es bei JUnit 4 und 5 unterschiedliche Objekt-/Referenz-Pooling-Mechanismen.

Reply:
Und zu (2), ich war's nicht schuld. 😅 ... Aber die Chest-Klasse gibt es schon so lange, ich weiß nicht mehr, wer diese ins Leben gerufen hatte. :)

Reply:
@cagix IntelliJ zeigt(e) `assertEquals` auch als "deprecated" an, aus eben genau dem Grund, weil Arrays `equals` nicht überschreiben (stattdessen wird dann `==` verwendet): https://stackoverflow.com/questions/41987634/why-is-assertequalsobject-object-from-junit-4-deprecated

... und `==` _kann_ sich eben wie ein richtig implementiertes `equals` verhalten, _muss_ dies jedoch nicht.

Und ich vermute einfach, dass der Aufruf von `inventoryComponent.get().items()` und `new Item[] {null, null, null, null, null, null, null, null, null, null, null, null}` bei JUnit 4 jeweils auf ein schon vorhandenes, mit `null` initialisiertes Array zurückgreift; das bei JUnit 5 jedoch nicht der Fall ist.

Reply:
> ... und `==` _kann_ sich eben wie ein richtig implementiertes `equals` verhalten, _muss_ dies jedoch nicht.

Habe grad keine Quellen, aber ich meine mich zu erinnern, dass `==` in Java **immer** der Vergleich der Objektreferenzen ist? `.equals` **kann** auf `==` zurückgeführt werden (sollte es aber nicht, aus verschiedenen Gründen). Oder habe ich das falsch im Gedächtnis?

> 
> Und ich vermute einfach, dass der Aufruf von `inventoryComponent.get().items()` und `new Item[] {null, null, null, null, null, null, null, null, null, null, null, null}` bei JUnit 4 jeweils auf ein schon vorhandenes, mit `null` initialisiertes Array zurückgreift; das bei JUnit 5 jedoch nicht der Fall ist.

Sorry, aber `new Item[]...` legt *ad hoc* ein neues Objekt (Array) an, und aus `inventoryComponent.get().items()` kommt auch ein Objekt - aber ein anderes (wenn auch leeres) Objekt/Array. Wie sollen die denn in Java jemals die selbe Objekt-ID haben? Bei Strings wird zwar im Hintergrund ein wenig Zauber gemacht, aber bei einfachen Arrays???

Reply:
@AMatutat Bitte gib mal zeitnah einen Kommentar ab, sonst ufert das hier wieder aus 🤣

Reply:
> > ... und `==` _kann_ sich eben wie ein richtig implementiertes `equals` verhalten, _muss_ dies jedoch nicht.
> 
> Habe grad keine Quellen, aber ich meine mich zu erinnern, dass `==` in Java **immer** der Vergleich der Objektreferenzen ist?

wahr ... das war ein Typo, ich glaube, ich meinte `assertEquals`, nicht `==` 😬

Reply:
> Sorry, aber `new Item[]...` legt _ad hoc_ ein neues Objekt (Array) an,

Eigentlich bin ich da bei dir, aber ich habe jetzt gelesen, dass `<Type>[]` ein "Zwischentyp" zwischen einem echten `Object` und einem primitive type sein kann: https://stackoverflow.com/questions/12806739/is-an-array-a-primitive-type-or-an-object-or-something-else-entirely

Aber in der JLS steht es eigentlich anders: https://docs.oracle.com/javase/specs/jls/se22/html/jls-10.html

> In the Java programming language, arrays **are** objects ([§4.3.1](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.1)), are dynamically created, and may be assigned to variables of type Object ([§4.3.2](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.2)).

Also, kurz gesagt, ich weiß es leider nicht genau.

Reply:
> > Sorry, aber `new Item[]...` legt _ad hoc_ ein neues Objekt (Array) an,
> 
> 
> 
> Eigentlich bin ich da bei dir, aber ich habe jetzt gelesen, dass `<Type>[]` ein "Zwischentyp" zwischen einem echten `Object` und einem primitive type sein kann: https://stackoverflow.com/questions/12806739/is-an-array-a-primitive-type-or-an-object-or-something-else-entirely
> 
> 
> 
> Aber in der JLS steht es eigentlich anders: https://docs.oracle.com/javase/specs/jls/se22/html/jls-10.html
> 
> 
> 
> > In the Java programming language, arrays **are** objects ([§4.3.1](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.1)), are dynamically created, and may be assigned to variables of type Object ([§4.3.2](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.2)).
> 
> 
> 
> Also, kurz gesagt, ich weiß es leider nicht genau.

stackoverflow ist für mich keine referenz.

Reply:
> > > Sorry, aber `new Item[]...` legt _ad hoc_ ein neues Objekt (Array) an,
> > 
> > 
> > 
> > Eigentlich bin ich da bei dir, aber ich habe jetzt gelesen, dass `<Type>[]` ein "Zwischentyp" zwischen einem echten `Object` und einem primitive type sein kann: https://stackoverflow.com/questions/12806739/is-an-array-a-primitive-type-or-an-object-or-something-else-entirely
> > 
> > 
> > 
> > Aber in der JLS steht es eigentlich anders: https://docs.oracle.com/javase/specs/jls/se22/html/jls-10.html
> > 
> > 
> > 
> > > In the Java programming language, arrays **are** objects ([§4.3.1](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.1)), are dynamically created, and may be assigned to variables of type Object ([§4.3.2](https://docs.oracle.com/javase/specs/jls/se22/html/jls-4.html#jls-4.3.2)).
> > 
> > 
> > 
> > Also, kurz gesagt, ich weiß es leider nicht genau.
> 
> stackoverflow ist für mich keine referenz. 

Das verlangt ja auch keiner, aber eine andere "Erklärung" für das "merkwürdige" Verhalten habe ich nicht gefunden, und wir sind uns ja einig, dass das asserEquals nicht gepasst hat. Ich kann nur vermuten, dass auch bei Arrays auf eine Art Heap zurückgegriffen wird, und so das new "übergangen" wird - oder dass die jvm hier spezielle Magie macht.

Reply:
@cagix Moin ☕ ...

Ich habe noch mal in den alten JUnit 4-Code geschaut ... Der Grund, weshalb der Test nicht fehlschlug, war, dass intern an `assertArrayEquals` delegiert wurde, was aber keine Referenzen vergleicht, sondern den Inhalt überprüft:

![grafik](https://github.com/user-attachments/assets/aa2d6430-0683-488b-8988-3a6b71f2f006)

In JUnit 5 findet dieses Delegieren aber nicht mehr statt. Deshalb wurde das `assertEquals` in JU 4 auch als `deprecated` gekennzeichnet.

Sorry noch einmal für die Verwirrung.

Reply:
das war die antwort, auf die ich gewartet habe 👌